### PR TITLE
[#1837563] Migrate to app service library to trigger functions

### PR DESCRIPTION
**Pull request by Flanker32**
Migrate to app service library to trigger functions

### CODE REVIEW

**Comment by Flanker32** [comment]
AB#1837563

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java [hunk 1]
```diff
@@ -179,32 +134,17 @@ private String getFunctionHttpTriggerUrl() {
...
+        return String.format(HTTP_TRIGGER_URL_WITH_CODE, functionApp.hostName(), this.name, functionApp.getMasterKey());
     }
 
-    private Map<String, String> listFunctionKeys() {
-        // todo: imply with app service library
-        return Collections.EMPTY_MAP;
+    @RequiredArgsConstructor
+    static class TriggerRequest {
+        private final String input;
     }
```
Comment (shenqianjin) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java:149]: can we user only String to replace this class ?

Comment (Flanker32) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java:149]: trigger api in sdk will serialize input object into json map, so we could not pass the value string directly which will miss the key

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java [hunk 2]
```diff
@@ -96,52 +90,13 @@ private void trigger() {
                 triggerHttpTrigger(trigger);
                 break;
             case "timertrigger":
-                triggerTimerTrigger();
-                break;
-            case "eventhubtrigger":
-                triggerEventHubTrigger();
+                functionApp.triggerFunction(this.name, new Object()); // no input for timer trigger
```
Comment (Flanker32) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java:93]: try replace with null

**Review by andxu — APPROVED** [review]

**Pull request merged**